pkg/reconciler/artifactbuild/artifactbuildrequest.go

```diff
@@ -191,7 +218,7 @@ func (r *ReconcileArtifactBuild) handleStateDiscovering(ctx context.Context, abr
 			SCMType: abr.Status.SCMInfo.SCMType,
 			Tag:     abr.Status.SCMInfo.Tag,
 			Path:    abr.Status.SCMInfo.Path,
-		}}
+		}, BuildRecipes: buildRecipes}
```

Comment:
my comment applies to the line below, which is unchanged, https://github.com/redhat-appstudio/jvm-build-service/pull/60/files#diff-e8e0091cd27c2b3a10a3f5ec14aa8c07390ca462d30847514323d45e2470ff6eR222 ... i've missed this detail previously as well.

just noticed we are calling status().update(), but we are manipulating the spec too

and this is actually working?  my guess is it is working, otherwise you would have ran into issues already, but at the lower levels of the k8s api, when you define a status subresource a la https://github.com/redhat-appstudio/jvm-build-service/blob/main/pkg/apis/jvmbuildservice/v1alpha1/dependencybuild_types.go#L41 and https://github.com/redhat-appstudio/jvm-build-service/blob/main/deploy/crds/base/jvmbuildservice.io_dependencybuilds.yaml#L175-L176 you have to call status.update to update the status and update when you update the spec/metadata

either controller-gen is not crafting status subresource correctly, or controller-gen's client must do some machinations to handle things regardless of which update you call

or I'm just misremembering how the lower level k8s update status works

maybe I investigate to satisfy my curiosity if you say "yes, it just works", but we'll see if time permits

Reply:
We are working on two different objects. We update the status of the AB and create a new DependencyBuild.